lock/README.md

```diff
@@ -0,0 +1,57 @@
+# lock
+
+## Prerequisites
+
+* [winglang](https://winglang.io).
+
+## Installation
+
+`sh
+npm i @winglibs/lock
+`
+
+## Usage
+
+```js
+bring lock;
+bring cloud;
+bring util;
+
+let l = new lock.Lock();
+let q = new cloud.Queue();
+
+q.setConsumer(inflight (m: str) => {
+  let id = util.uuidv4();
+  l.acquire("my-lock", 20s);
```

Comment:
What happens if a function crashes while it is holding the lock? Does the hold on the lock have an automatic expiry of some kind? Without some automatic expiry or heartbeat, I'm not sure if this resource would be useful in production.